### ISSUE TITLE
BYOP

### ISSUE THREAD

**Issue by jmeas** [QubitProducts/cherrytree#85]
This lib. shouldn't force you to use a particular Promise implementation imo. It's slowly becoming less and less necessary to polyfill Promises for two reasons:

1. The # of users on modern browsers is very large, and steadily increasing
2. They're such a common feature that many (most?) devs already have a Promise implementation of choice.

In addition, unless Cherrytree depends on Promises to load up (it doesn't seem to, but correct me if I'm wrong!) then there's an opportunity for a developer to specify what Promise lib they want to use before they start using Cherrytree.

BYOP in the library may look like:

```js
if (typeof Promise !== 'undefined') {
  Cherrytree.Promise = Promise;
}
```

in addition to a note in the README.

Pros/cons:

**Pros:**

- Developers can choose what Promise lib they want to use
- For most devs this will make Cherrytree a smaller dep

**Cons:**

- A bit more work to set up Cherrytree in legacy environments/envs where you're not using the global promise obj (like if you're using RSVP, for instance)

**Comment by KidkArolis** [comment]
Yes, this is definitely a good idea.

**Comment by KidkArolis** [comment]
Fixed via #88
